clients/cobol-lsp-vscode-extension/src/__tests__/task/CustomBuildTaskTerminalTest.spec.ts

```diff
@@ -0,0 +1,224 @@
+        expect(mockCompileFunction).toBeCalledTimes(1);
+    });
+
+    it("returns with error when ZE is not installed", async () => {
+        (vscode.extensions as any) = {
+            getExtension : jest.fn().mockReturnValue(undefined),
+        };
+
+        await (taskTerminal as any).doCompile().catch(e => {
+            expect(e).toEqual("Please install zowe Explorer");
+        });
+    });
+
+    it("returns with error when ZE profile is not found", async () => {
+        vscode.workspace.getConfiguration = jest.fn().mockReturnValue({
+            get: jest.fn().mockReturnValue(""),
+        });
+        await (taskTerminal as any).doCompile().catch(e => {
+            expect(e).toEqual("Please specify zowe profile for copybook download for the task to run.");
+        });
+    });
+
+    it("return with error when jobCard and steplib is not provided", async () => {
+        const incompleteTask = new CustomBuildTaskTerminal({ fileName: "filename" } as any, {
+            type: "testTask",
+        });
+        await (incompleteTask as any).doCompile().catch(e => {
+            expect(e).toEqual("jobCard is mandatory. customize task and provide these attributed in task.json");
+        });
+    });
+
+    it("checks compile works when all data is provided", async () => {
+        vscode.workspace.getConfiguration = jest.fn().mockReturnValue({
+            get: jest.fn().mockReturnValue("testProfile"),
+        });
+        (taskTerminal as any).CompileCobolDocument = jest.fn().mockResolvedValue("TEST");
+        await expect((taskTerminal as any).doCompile()).resolves.not.toThrowError();
+    });
+
+    it("checks proper flow of task", async () => {
+        const loadedProfile: any = {};
+        const task = new CustomBuildTaskTerminal({ fileName: "filename" } as any, {
+            jobCard,
+            steplib: ["steplib1", "steplib2"],
+            syslib: ["syslib1", "syslib2"],
+            type: "testTask",
+        });
+        (task as any).storeSysPrint = jest.fn();
+        (task as any).getCompileJobJCL = jest.fn().mockReturnValue("JCL TO SUBMIT");
+        (task as any).documentText = {
+            getText: jest.fn().mockReturnValue("THIS IS A DUMMY COBOL PRGM."),
+            uri: {
+                fsPath: "documentUri",
+            },
+        };
+        (task as any).writeEmitter = {
+            fire: writterEvent,
+        };
+
+        (task as any).closeEmitter = {
+            fire: closeEvent,
+        };
+        await (task as any).CompileCobolDocument(vscode.extensions.getExtension("Zowe.vscode-extension-for-zowe")?.exports, loadedProfile);
+        expect(writterEvent).toBeCalledTimes(6);
+    });
+
+    it("checks proper JCL is formed for compile job", async () => {
+        jobcardFilePath = path.resolve(vscode.workspace.workspaceFolders[0].uri.fsPath, jobCard);
+        await fs.writeFileSync(jobcardFilePath, "JOBCARD");
+
+        const expectedJCL = "JOBCARD\r\n" +
```

Comment:
Why do some lines contain '\r\n' and some '\n' only.

Reply:
Used "os" node module. But the line separator coming from the compile scripts are shown as \n and by EOL as \r\n on windows

Reply:
my mistake, updated